### PR TITLE
Calculations

### PR DESCRIPTION
**Pull request by hannibal00212**
Add support for calculations in neu search bar directly and changed the formatting to ignore zero on the right side of comma
image

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/util/Calculator.java [hunk 1]
```diff
@@ -19,19 +19,44 @@
...
+				BigDecimal calculate = calculate(input);
+				lastResult = new DecimalFormat("#,##0.##").format(calculate);
+			} catch (CalculatorException ignored) {
+				lastResult = null;
+			}
+		}
+
+		return lastResult;
+	}
+
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/util/Calculator.java:59]: I would prefer to keep Calculator.java a pure calculator, untainted by minecraft business logic; therefore id prefer if wed move this to another class (perhaps to the callsite, or to another place)

**Review by romangraef — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/util/Calculator.java (hunk 1 above)
Comment (hannibal00212) [reply src/main/java/io/github/moulberry/notenoughupdates/util/Calculator.java:59]: Actually a good idea. I prefer not having the fields in there too. Ill move them.

**Review by romangraef — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Comment by NopoTheGamer** [comment]
[I cant see anything I type](https://cdn.discordapp.com/attachments/832652653292027904/1005384248733212752/unknown.png) and it shows null when nothing is typed

**Review by NopoTheGamer — CHANGES_REQUESTED** [review]
i think i was meant to click the request changes button

**Comment by hannibal00212** [comment]
I can reproduce your behaviour. Funny that both errors are because of missing brackets. Hopefully, this is the last time I push changes without testing them—no matter how small they are.

**Review by NopoTheGamer — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**